usaspending_api/recipient/v2/views/list_recipients.py

```diff
@@ -1,59 +1,60 @@
-import logging
 import copy
+import logging
 
+from django.db.models import F, Q
 from rest_framework.response import Response
 from rest_framework.views import APIView
-from django.db.models import F, Q
 
 from usaspending_api.common.cache_decorator import cache_response
 from usaspending_api.common.helpers.generic_helper import get_pagination_metadata
 from usaspending_api.common.validator.pagination import PAGINATION
 from usaspending_api.common.validator.tinyshield import TinyShield
 from usaspending_api.common.validator.utils import update_model_in_list
 from usaspending_api.recipient.models import RecipientProfile
-from usaspending_api.recipient.v2.lookups import SPECIAL_CASES
+from usaspending_api.recipient.v2.lookups import AWARD_TYPES, SPECIAL_CASES
 
 logger = logging.getLogger(__name__)
 
 
-# based from all_award_types_mappings in usaspending_api.awards.v2.lookups.lookups
-AWARD_TYPES = {
-    "contracts": {"amount": "last_12_contracts", "filter": "contract"},
-    "grants": {"amount": "last_12_grants", "filter": "grant"},
-    "direct_payments": {"amount": "last_12_direct_payments", "filter": "direct payment"},
-    "loans": {"amount": "last_12_loans", "filter": "loans"},
-    "other_financial_assistance": {"amount": "last_12_other", "filter": "other"},
-}
-
-
-def get_recipients(filters={}):
-    lower_limit = (filters["page"] - 1) * filters["limit"]
-    upper_limit = filters["page"] * filters["limit"]
-
+def build_duns_base_query(filters):
     qs_filter = Q()
     if "keyword" in filters:
         qs_filter |= Q(recipient_name__contains=filters["keyword"].upper())
         qs_filter |= Q(recipient_unique_id__contains=filters["keyword"])
 
+    if filters["award_type"] != "all":
+        qs_filter &= Q(award_types__overlap=[AWARD_TYPES[filters["award_type"]]["filter"]])
+
+    return qs_filter
+
+
+def get_recipients(filters={}, count=None):
+    lower_limit = (filters["page"] - 1) * filters["limit"]
+    upper_limit = filters["page"] * filters["limit"]
+
     amount_column = "last_12_months"
     if filters["award_type"] != "all":
         amount_column = AWARD_TYPES[filters["award_type"]]["amount"]
-        qs_filter &= Q(award_types__overlap=[AWARD_TYPES[filters["award_type"]]["filter"]])
+
+    qs_filter = build_duns_base_query(filters)
 
     queryset = (
         RecipientProfile.objects.filter(qs_filter)
         .values("recipient_level", "recipient_hash", "recipient_unique_id", "recipient_name", amount_column)
         .exclude(recipient_name__in=SPECIAL_CASES)
     )
-
     api_to_db_mapper = {"amount": amount_column, "duns": "recipient_unique_id", "name": "recipient_name"}
 
+    nulls_last = filters["sort"] in ["name", "duns"]
```

Comment:
Commented: 

> Nulls Last isn't enabled for the amount sort because it prevents queries sorted by amount columns DESC from using an index on those columns, even though they do cannot contain nulls

Reply:
Commented: 

> Nulls Last isn't enabled for the amount sort because it prevents queries sorted by amount columns DESC from using an index on those columns, even though they do cannot contain nulls